app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -471,7 +471,7 @@ def construct_repo_from_params(repo_params)
                                                             :gpg_key, :ssl_ca_cert, :ssl_client_cert, :ssl_client_key,
                                                             :checksum_type, :download_policy).to_h.with_indifferent_access)
       root.docker_upstream_name = repo_params[:docker_upstream_name] if repo_params[:docker_upstream_name]
-      root.docker_tags_whitelist = repo_params[:docker_tags_whitelist] if repo_params[:docker_tags_whitelist]
+      root.docker_tags_whitelist = repo_params.fetch(:docker_tags_whitelist, []) if root.docker?
```

Comment:
do you feel this is something that can be better accomplished in the model layer instead of controller ?

Reply:
This method clearly needs to be refactored (and probably moved to the model), as seen by the `rubocop:disable Metrics/CyclomaticComplexity`. I opted to change it as little as possible. IMO I haven't reduced or increased tech debt with this change.

Reply:
Ok I missed the more important bug here
```
root.docker_tags_whitelist =  repo_params.fetch(:docker_tags_whitelist, []) if root.docker?
```
What if I sent an update api call changing just the feed url. Will this change overwrite existing docker tags.?

Reply:
Actually, this code isn't run for update, just create, so I think it's fine as is. I just tried the scenario you suggest, and it works as expected.

Reply:
good to know :)

Reply:
@akofink errr, its called for update too?    https://github.com/Katello/katello/blob/1a240a5507361e3e10c0cc4a56b3f30180c1b1c8/app/controllers/katello/api/v2/repositories_controller.rb#L312

Reply:
@jlsherrill aren't we talking about the `construct_repo_from_params` method?

Reply:
oh sorry, shouldn't code review after migraine ;)